kivy/loader.py

```diff
@@ -122,6 +124,13 @@ def __init__(self):
         self._start_wanted = False
         self._trigger_update = Clock.create_trigger(self._update)
 
+        try:
+            assert platform == 'android'
```

Comment:
I'm really curious about the reason of this construction, that seems exactly equivalent to the more straightforward `if platform == 'android'`, am i missing something?

Reply:
try/except is generally considered more "pythonic" and should be used instead of conditional statements whenever possible. I read it in a book, but don't recall which book. It has something to do with the fundamental philosophy of Python; along the lines of "it's better to try and fail, then to ask permission" - to paraphrase

Reply:
I think it's a common myth in the python community, and it's been addressed elsewhere 
https://www.reddit.com/r/Python/comments/3o36h8/would_it_be_better_to_use_tryexcept_rather_than/ points to 
https://mail.python.org/pipermail/python-dev/2014-March/133118.html
as reference, while not of biblical importance, Guido weighted on that, and i think we can consider a good reference on the philosophy of Python :).
> and I disagree with the position that EAFP is better than
LBYL, or "generally recommended" by Python.

there are situations where it's the best tool, and it's there to be used (especially when we don't have the easy possiblity to check for error conditions before calling the code). But here you use assert to do the test right after, so i don't think it's pythonic at all, but thanks for trying, i appreciate the intention, even if it comes from a misconception.

After this is switched to a simple test, I think i'd be fine with merging it, i'll update my own PR to p4a to make sure we always have certifi in kivy on android. Thanks for putting up with my feedback :).

Reply:
> I think it's a common myth in the python community, and it's been addressed elsewhere
> https://www.reddit.com/r/Python/comments/3o36h8/would_it_be_better_to_use_tryexcept_rather_than/ points to
> https://mail.python.org/pipermail/python-dev/2014-March/133118.html
> as reference, while not of biblical importance, Guido weighted on that, and i think we can consider a good reference on the philosophy of Python :).
> 
> > and I disagree with the position that EAFP is better than
> > LBYL, or "generally recommended" by Python.
> 
> there are situations where it's the best tool, and it's there to be used (especially when we don't have the easy possiblity to check for error conditions before calling the code). But here you use assert to do the test right after, so i don't think it's pythonic at all, but thanks for trying, i appreciate the intention, even if it comes from a misconception.
> 
> After this is switched to a simple test, I think i'd be fine with merging it, i'll update my own PR to p4a to make sure we always have certifi in kivy on android. Thanks for putting up with my feedback :).

I agree it is mostly aesthetic preference. But what about handling the hypothetical edge case of a missing certifi when it is running on Android? Wouldn't the try/except (with an additional except block to handle missing a certifi) be necessary, anyways?

What does a missing certifi error look like? Would it be a missing .pem or module not found?

Or should there be no concern for a potential missing certifi on Android, and so we should just use conditional statements?

Reply:
it would be a `ModuleNotFoundError`, in such case yeah we could totally try except for that in all cases, that's the alternative solution i was thinking about earlier, to me either/or, check platform and import, or try import and if it just silently ignore it, maybe with a warning in log, but that feels less clean than checking.

The current way (commit ab63037) seems clean to me, thanks.

Reply:
So I suppose this is finished. Thank you, you have been super helpful throughout this process, and I am thankful that the Kivy developers have provided such a powerful and comprehensive framework supporting portable gui applications with Python.

I will now attempt to integrate GPS and permissions fixes to plyer (which are very much needed!) (This might take awhile!)

Reply:
> it would be a `ModuleNotFoundError`, in such case yeah we could totally try except for that in all cases, that's the alternative solution i was thinking about earlier, to me either/or, check platform and import, or try import and if it just silently ignore it, maybe with a warning in log, but that feels less clean than checking.
> 
> The current way (commit [ab63037](https://github.com/kivy/kivy/commit/ab63037d0d35f2b16abaef016e0bb86970349351)) seems clean to me, thanks.

Do I need to re-request review for matham?